app/assets/javascripts/application.js

```diff
@@ -17,11 +17,84 @@
 //= require turbolinks
 //= require_tree .
 
+(function ($) {
+    $.fn.longClick = function (callback, timeout) {
+        var timer;
+        timeout = timeout || 500;
+        $(this).mousedown(function () {
+            timer = setTimeout(function () {
+                callback();
+            }, timeout);
+            return false;
+        });
+        $(document).mouseup(function () {
+            clearTimeout(timer);
+            return false;
+        });
+    };
 
-function addRows(count) {
-    for (i = 0; i < count; i++) {
-        $(".card .content").append('<div class="action" id="action_' + i + '">Slot ' + (i + 1) + '</div>');
-    }
+})(jQuery);
+
+
+function queryApi(date) {
+    $.ajax({
+        dataType: "json",
+        url: '/api_index.json?date=' + date,
+        success: function (response) {
+            loadCurrentBookings(response);
+            addEvents();
+        }
+    });
+}
+
+function updateApi(date) {
+    $.ajax({
+        dataType: "json",
+        url: '/api_index.json?date=' + date,
+        success: function (response) {
+            updateCurrentBookings(response);
+            addEvents();
+        }
+    });
+}
+
+
+function loadCurrentBookings(response) {
+    var res = response;
+    res.items.forEach(function (item) {
+        var card = ['#card', item.id].join('-');
+        $(card + " .content").append('<div class="with-scroll"></div>')
+        item.slots.forEach(function (slot) {
+            $(card + " .content .with-scroll").append('<div class="action" id="action_' + slot.info.id + '" style="background-color: ' + getBackgroundColor(slot) + '">' + getInfo(slot) + '</div>');
+        });
+    });
+}
+
+function updateCurrentBookings(response) {
+    var res = response;
+    res.items.forEach(function (item) {
+        var card = ['#card', item.id].join('-');
+        $(card + " .content .with-scroll").empty();
+        item.slots.forEach(function (slot) {
+            $(card + " .content .with-scroll").append('<div class="action" id="action_' + slot.info.id + '" style="background-color: ' + getBackgroundColor(slot) + '">' + getInfo(slot) + '</div>');
+        });
+    });
+
+}
+
+function getBackgroundColor(obj) {
+    var color = (obj.state == 'booked') ? 'red' : 'green';
+    return color;
+}
+
+function getInfo(obj) {
+    var message = (obj.info.client != null) ? setSlotMessage(obj) : obj.info.time;
+    return message;
+}
+
+function setSlotMessage(obj) {
+    var message = [obj.info.time, 'Grupp: ' + obj.info.client, 'Start: ' + obj.info.booking_time.split(' - ')[0], 'Slut: ' + obj.info.booking_time.split(' - ')[1]].join(' ');
```

Comment:
Why not splitting the `booking_time` once instead of twice?
```
var booking_times = obj.info.booking_time.split(' - ')
var message = [obj.info.time, 'Grupp: ' + obj.info.client, 'Start: ' + booking_times[0], 'Slut: ' + booking_times[1]].join(' ');
```